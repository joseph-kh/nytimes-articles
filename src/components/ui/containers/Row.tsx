import React from "react";
import { StyleSheet, View as ViewElement, ViewProps } from "react-native";

const Row: React.FC<ViewProps> = (props) => {
  const { style, children } = props;

  return (
    <ViewElement style={StyleSheet.flatten([styles.centerContainer, style])}>
      {children}
    </ViewElement>
  );
};

const styles = StyleSheet.create({
  centerContainer: {
    flexDirection: "row",
    alignItems: "center",
  },
});

export default Row;
