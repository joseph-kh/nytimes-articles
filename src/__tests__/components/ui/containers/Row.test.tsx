import React from "react";
import { create } from "react-test-renderer";
import Row from "@/components/ui/containers/Row";

describe("Row Component", () => {
  it("renders correctly", () => {
    const tree = create(<Row>Hello, World!</Row>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("applies custom styles", () => {
    const customStyle = { backgroundColor: "red" };
    const tree = create(<Row style={customStyle}>Hello, World!</Row>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
