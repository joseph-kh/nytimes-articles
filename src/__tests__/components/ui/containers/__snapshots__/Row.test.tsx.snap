// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row Component applies custom styles 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "red",
      "flexDirection": "row",
    }
  }
>
  Hello, World!
</View>
`;

exports[`Row Component renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  Hello, World!
</View>
`;
